src/aspire/utils/preprocess.py

```diff
@@ -302,3 +303,34 @@ def vol2img(volume, rots, L=None, dtype=None):
         im = im * m_reshape(np.exp(2*np.pi*1j*(grid2d['x'] +grid2d['y'])/(2*lv)), (lv, lv, 1))
 
     return np.real(im)
+
+
+def fuzzy_mask(L, r0, risetime, origin=None):
+    """
+    Create a centered 1D to 3D object of radius r0
+
+    Made with an error function with effective rise time.
+    :param L: The size of image
+    :param r0: The specified radius
+    :param risetime: The rise time for `erf` function
+    :param origin: The coordinates of origin
+    :return: The desired fuzzy mask
+    """
+    if isinstance(L, int):
+        L = np.array(L)
+    if isinstance(r0, int):
+        r0 = np.array(r0)
+
+    center = np.floor(L / 2)
+    if origin is None:
+        origin = center
+        origin = origin.astype('int')
+
+    grids = [np.arange(1-org, ell - org + 1) for ell, org in zip(L, origin)]
+    XYZ = np.meshgrid(*grids, indexing='ij')
+    XYZ_sq = [X ** 2 for X in XYZ]
+    R = np.sqrt(np.add(*XYZ_sq))
```

Comment:
but np.add only takes two args, so you are only getting np.add(XYZ_sq[0], XYZ_sq[1]).  Is that what you wanted for 3d?

If not you maybe meant, `np.sum(XYZ_sq, axis=0)`? Or , I'm not even sure what you intended, that's why this is troubling.. If you really intend to throw out the last `XYZ_sq` write `np.add(XYZ_sq[0], XYZ_sq[1])` or something...

Reply:
You are right. Done in commit 03944b1.

Reply:
Cool, thanks for fixing it.